libminifi/include/utils/TimeUtil.h

```diff
@@ -70,6 +72,16 @@ class Clock {
  public:
   virtual ~Clock() = default;
   virtual std::chrono::milliseconds timeSinceEpoch() const = 0;
+  virtual std::chrono::milliseconds wait_until(std::condition_variable& cv, std::unique_lock<std::mutex>& lck, std::chrono::milliseconds time, const std::function<bool()>& pred) {
+    auto now = timeSinceEpoch();
+    if (now < time) {
+      cv.wait_for(lck, time - now, [&] {
+        now = timeSinceEpoch();
+        return now >= time || pred();
+      });
+    }
+    return timeSinceEpoch();
```

Comment:
I think it would be better to tell in the return value that a timeout happened or the predicate returned true.

Reply:
yes it makes sense to be inline with the interface of `std::condition_variable::wait_until`, changed it to return a `bool`